### PR TITLE
Refactor the Tendermint module to run it on a single thread

### PR DESCRIPTION
**Pull request by sgkim126**

### CODE REVIEW

File: core/src/consensus/tendermint/mod.rs [hunk 1]
```diff
@@ -1197,29 +1471,50 @@ impl ConsensusEngine<CodeChainMachine> for Tendermint {
...
+            .unwrap();
+        receiver.recv().unwrap()
     }
 
     /// Called when the node is the leader and a proposal block is generated from the miner.
     /// This writes the proposal information and go to the prevote step.
     fn proposal_generated(&self, sealed_block: &SealedBlock) {
-        let mut guard = self.inner.lock();
-        guard.proposal_generated(sealed_block);
+        // TODO
```
Comment (majecty) [inline core/src/consensus/tendermint/mod.rs]: What is the purpose of the TODO?

File: core/src/consensus/tendermint/mod.rs [hunk 2]
```diff
@@ -1273,74 +1568,101 @@ impl ConsensusEngine<CodeChainMachine> for Tendermint {
...
+            })
+            .unwrap();
+        receiver.recv().unwrap()
     }
 
     fn set_signer(&self, ap: Arc<AccountProvider>, address: Address) {
-        let mut guard = self.inner.lock();
-        guard.set_signer(ap, address)
+        self.has_signer.store(true, AtomicOrdering::SeqCst);
+        // TODO
```
Comment (majecty) [inline core/src/consensus/tendermint/mod.rs]: What is the purpose of the TODO?

File: core/src/consensus/tendermint/mod.rs (hunk 1 above)
Comment (sgkim126) [reply core/src/consensus/tendermint/mod.rs]: Sorry. It should be removed. I'll update the commit.

File: core/src/consensus/tendermint/mod.rs (hunk 2 above)
Comment (sgkim126) [reply core/src/consensus/tendermint/mod.rs]: ditto

**Pull request merged**